cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitutionSPO.hs

```diff
@@ -233,3 +246,43 @@ hprop_ledger_events_propose_new_constitution_spo = H.integrationWorkspace "propo
 
   exitCode H./== ExitSuccess -- Dit it fail?
   H.assert $ "DisallowedVoters" `isInfixOf` stderr -- Did it fail for the expected reason?
+
+checkNewConstitutionHasBeenProposed ::
+  (HasCallStack, MonadIO m, MonadTest m)
+  => NodeConfigFile In
+  -> SocketPath
+  -> m (Maybe (L.GovActionId StandardCrypto))
+checkNewConstitutionHasBeenProposed nodeConfigFile socketPath = do
+  result <- runExceptT $ checkLedgerStateCondition nodeConfigFile socketPath QuickValidation
+    (EpochNo maxBound) Nothing
```

Comment:
You might want to put something smaller instead of `maxBound`, to make the test fail sooner, rather than looping infinitely waiting on a condition.

Reply:
Done :+1: